File: spring_java_orm_sample/src/main/java/com/example/demo/config/SearchProperties.java
package com.example.demo.config;

import jakarta.validation.constraints.Min;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;
import org.springframework.validation.annotation.Validated;

@Component
@ConfigurationProperties(prefix = "search")
@Validated
@Getter
@Setter
public class SearchProperties {
    @Min(1)
    private int pageSize;
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/config/SearchProperties.kt
package com.example.demo.config

import jakarta.validation.constraints.Min
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component
import org.springframework.validation.annotation.Validated

@Component
@ConfigurationProperties(prefix = "search")
@Validated
class SearchProperties {
    @field:Min(1)
    var pageSize: Int = 0
}
